### PR TITLE
Improve error emitting in bad xml parsing

### PR DESCRIPTION
**Pull request by pgilad**

### CODE REVIEW

**Comment by adam-lynch** [comment]
This just uses Gulp's error reporting instead yeah? Does the task die then here? Are you sure through2 doesn't keep going? Or maybe it should, I don't know.

**Comment by pgilad** [comment]
This will fail the task unless the stream error is caught (using `.on('error')`.

The reason for this change, is that the cb is already called at the end of the block, and callbacks should only be called once. This will have the exact same effect, but won't call the callback twice (which is an error).

**Comment by pgilad** [comment]
Make sure the test is passing, I don't know why the travis build hasn't completed :question:

**Comment by adam-lynch** [comment]
Sounds good. Yeah I saw that but it seems the two builds actually did finish (sucessfully). I'll look into it.

**Comment by pgilad** [comment]
Travis now does it's job..

**Comment by adam-lynch** [comment]
But now I can't merge because you've undone some of your code?

**Comment by pgilad** [comment]
Sure you can, you merge the branch, it just contains 2 commits. Worst cast - do it manually

File: index.js [hunk 1]
```diff
@@ -37,7 +37,10 @@ module.exports = function(siteUrl){
 
         var self = this;
         parseXML(xml, function (err, result) {
-            if(err) return cb(new PluginError(pluginName, err));
+            if(err) {
+                cb(new PluginError(pluginName, err));
```
Comment (adam-lynch) [inline index.js:41]: I'm saying that as it stands right now, the branch doesn't contain what you wanted. It doesn't emit an error event.

**Comment by pgilad** [comment]
Running `cb(error)` is the same as emitting the error. It's just important to call the callback once, and there are 2 ways to call it:
- If parsing XML failed
- If parsing XML succeded

Bedies that - we push the files parsed from XML, but still call the cb only once.

**Comment by adam-lynch** [comment]
Ok, thanks for explaining :+1:

**Comment by adam-lynch** [comment]
`1.0.2` published

**Pull request merged**